skimage/restoration/_denoise.py

```diff
@@ -376,21 +390,50 @@ def _wavelet_threshold(img, wavelet, threshold=None, sigma=None, mode='soft'):
            DOI: 10.1093/biomet/81.3.425
 
     """
-    coeffs = pywt.wavedecn(img, wavelet=wavelet)
-    detail_coeffs = coeffs[-1]['d' * img.ndim]
+    if not isinstance(wavelet, pywt.Wavelet):
```

Comment:
Okay,  will remove the check.  I had not noticed that `wavelet` was already specified as a string rather than a `Wavelet` object in the docstring. 

It is now necessary to have the object representation so I can access the `dec_len` attribute of the wavelet object when determining the maximum possible levels of wavelet decomposition.